container/service_options.go

```diff
@@ -0,0 +1,131 @@
+package container
+
+import (
+	"github.com/docker/docker/api/types/mount"
+	"github.com/docker/docker/api/types/swarm"
+)
+
+// ServiceOptions is a simplify version of swarm.ServiceSpec that can be merge to it
+type ServiceOptions struct {
+	Image       string
+	Namespace   []string
+	Ports       []Port
+	Mounts      []Mount
+	Env         []string
+	Args        []string
+	NetworksID  []string
+	Labels      map[string]string
+	ServiceSpec swarm.ServiceSpec
+}
+
+// Port is a simplify version of swarm.PortConfig
+type Port struct {
+	Target    uint32
+	Published uint32
+}
+
+// Mount is a simplify version of mount.Mount
+type Mount struct {
+	Source string
+	Target string
+}
+
+func (options *ServiceOptions) merge() {
```

Comment:
I think if we want to have a proxy for these data it's fine but in that case we shouldn't add the `ServiceSpec` and like that this function is just a way to build the `ServiceSpec` and there is no "merge" which in most of cases are not even merge but more override so for me all this file doesn't really make sense. It's like trying to do something but half way, either we keep the original object (the docker one) or we create our own but not something in the middle especially because this add a lot more unnecessary complexity and we have to deal with these "merge" I really don't like that